### PR TITLE
Update eeveelution correction code to use the Pokemon Type

### PR DESCRIPTION
**Pull request by TripSixes**
This update implements the scanning of Pokemon Type so that the
information is available when we need to figure out which eeveelution
has been scanned.  Since each eeveelution has its own type, this
offers an accurate method for detecting each evolved form.

### CODE REVIEW

**Comment by TripSixes** [comment]
resolves #663

File: app/src/main/java/com/kamron/pogoiv/OcrHelper.java [hunk 1]
```diff
@@ -391,6 +391,30 @@ private String getPokemonNameFromImg(Bitmap pokemonImage) {
...
+        Bitmap type = getImageCrop(pokemonImage, 0.365278, 0.621094, 0.308333, 0.035156);
+        String hash = "type" + hashBitmap(type);
+        String pokemonType = ocrCache.get(hash);
+
+        if (pokemonType == null) {
+            type = replaceColors(type, true, 68, 105, 108, Color.WHITE, 200, true);
+            tesseract.setImage(type);
+            pokemonType = fixOcrNumsToLetters(tesseract.getUTF8Text().replace(" ", ""));
+            if (pokemonType.toLowerCase().contains("nidora")) {
+                pokemonType = getNidoranGenderName(pokemonImage);
```
Comment (nahojjjen) [inline app/src/main/java/com/kamron/pogoiv/OcrHelper.java]: Why is this code part of the type-check?

Comment (nahojjjen) [inline app/src/main/java/com/kamron/pogoiv/OcrHelper.java]: Just row 409

**Comment by TripSixes** [comment]
error by copy and paste.. will cleanup and resubmit.

**Comment by nahojjjen** [comment]
It looks like in githubs "files changed" that this RP now includes all changes that's been merged since you first made this PR, what happened?

**Comment by TripSixes** [comment]
I guess I made a mistake with the rebase.  Got any ideas how I quickly back
that​ out?

**Comment by nahojjjen** [comment]
From testing this branch, it just looks like the "changes" listed are the other branches that have been merged since this pr was started, and nothing seems to be missing. (?)

I think it's safe to merge this...

**Comment by TripSixes** [comment]
I would think that it would be safe to merge.. it should see each commit as already merged and move on.

**Pull request merged**